internal/pkg/template/pipeline.go

```diff
@@ -50,6 +50,7 @@ func withPipelineParsingFuncs() ParseOption {
 				return ok
 			},
 			"logicalIDSafe": ReplaceDashesFunc,
+			"alphanumeric":  StripNonAlphaNumFunc,
```

Comment:
What do you think of changing the `logicalIDSafe` function and still uses it in the template, instead of adding an `alphanumeric`?

Maybe we can have something like this:
```go
"logicalIDSafe": func (raw string) string {
   return StripNonAlphaNumFunc(ReplaceDashesFunc(raw))
},
```

Reply:
I don't love `DASH` and also worried about the effects on existing templates that have the old `logicalIDSafe` function.... But I can change it if we think that's better.

Reply:
>  also worried about the effects on existing templates

Only `BuildTestCommands` uses that function! That was also the reason why I suggested combining `ReplaceDashes` and `StripNonAlphaNum`, instead of just replacing `ReplaceDashs` with `StripNonAlphaNum`. It shouldn't have any effect on existing templates, because people wouldn't have been able to deploy the template anyway if the `BuildTestCommands` contains non-alphanum characters.

I guess one benefit of `ReplaceDahes` is that it keeps `frontend` and `front-end` two different identifiers... But I sincerely doubt if it's common for folks to have two identifiers whose only difference is on the dash 😂 

I don't think this is a huge deal though, just one nit! Not a huge fan of `DASH` either, tbh 😂  I am fine with keeping them separated!